sdk/eventhub/event-hubs/src/util/connectionUtils.ts

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { ConnectionEvents } from "rhea-promise";
+import { ConnectionContext } from "../connectionContext";
+import { logger } from "../log";
+
+/**
+ * Resolves once the context's connection emits a 'disconnected' event.
+ * @internal
+ * @ignore
+ * @param context
+ */
+export async function waitForConnectionDisconnected(context: ConnectionContext): Promise<void> {
+  return new Promise((resolve) => {
```

Comment:
Any reason this can't be a member function on the ConnectionContext?

Reply:
It could be. It's currently not just because this is what I started with, then later realized I needed to add things to the ConnectionContext, but I never touched this again.

Reply:
I agree with @richardpark-msft. In the below code, it looks odd that one is on the context and another is a helper that takes the context

<img width="614" alt="Screen Shot 2020-05-29 at 7 27 50 PM" src="https://user-images.githubusercontent.com/16890566/83317602-c64f7b80-a1e2-11ea-9f8a-086014f815b3.png">

Reply:
Yes, I agree. Will update.